app/src/main/java/com/sun/mywallpaper/ui/search/SearchFragment.kt

```diff
@@ -0,0 +1,134 @@
+    private lateinit var pagerAdapter: PagerAdapter
+    private val inputMethodManager by lazy {
+        (activity?.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager)
+    }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        if (context is OnSearchFragmentInteractionListener) {
+            listener = context
+        } else {
+            throw RuntimeException("$context $ERROR_IMPLEMENT_FRAGMENT_INTERACTION_LISTENER")
+        }
+    }
+
+    override fun initComponents() {
+        (activity as AppCompatActivity).apply {
+            setSupportActionBar(toolBar)
+            supportActionBar?.setDisplayHomeAsUpEnabled(true)
+            supportActionBar?.title = Constants.EMPTY_STRING
+        }
+        setHasOptionsMenu(true)
+
+        editTextSearch.apply {
+            requestFocus()
+            inputMethodManager.toggleSoftInput(InputMethodManager.SHOW_FORCED, 0)
+        }
+
+
+        pagerAdapter = PagerAdapter(childFragmentManager)
+        pagerAdapter.apply {
+        }
+        viewPager.apply {
+            adapter = pagerAdapter
+            addOnPageChangeListener(this@SearchFragment)
+            offscreenPageLimit = 2
+        }
+
+        tabLayout.setupWithViewPager(viewPager)
+
+        context?.let {
+            val spinnerAdapter =
+                ArrayAdapter.createFromResource(it, R.array.search_options, R.layout.item_spinner)
+            spinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+            spinnerSearchOption.apply {
+                adapter = spinnerAdapter
+                onItemSelectedListener = this@SearchFragment
+            }
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.search, menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean =
+        when (item.itemId) {
+            android.R.id.home -> {
+                onBackPressed()
+                true
+            }
+
+            R.id.actionClearText -> {
+                editTextSearch.text = null
+                editTextSearch.clearFocus()
+                inputMethodManager.hideSoftInputFromWindow(view?.windowToken, 0)
+                true
+            }
+
+            else -> super.onOptionsItemSelected(item)
+        }
+
+    override fun onBackPressed() = getNavigationManager().navigateBack()
+
+    override fun onPageScrollStateChanged(state: Int) {
+    }
+
+    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
+    }
+
+    override fun onPageSelected(position: Int) {
+        when (position) {
+            SEARCH_PHOTO_PAGE -> spinnerSearchOption.visibility = View.VISIBLE
+            SEARCH_COLLECTION_PAGE -> spinnerSearchOption.visibility = View.GONE
+            SEARCH_USER_PAGE -> spinnerSearchOption.visibility = View.GONE
+        }
```

Comment:
anh cos thể viết như này 
```suggestion
       spinnerSearchOption.visibility = when (position) {
            SEARCH_PHOTO_PAGE -> View.VISIBLE
            SEARCH_COLLECTION_PAGE -> View.GONE
            SEARCH_USER_PAGE -> View.GONE
        }
```

Reply:
ok